uber/site_sections/jobs.py

```diff
@@ -22,20 +10,40 @@ def job_dict(job, shifts=None):
         'restricted': job.restricted,
         'timespan': custom_tags.timespan.pretty(job),
         'location_label': job.location_label,
-        'shifts': shifts or [shift_dict(shift) for shift in job.shifts]
+        'shifts': [{
+            'id': shift.id,
+            'rating': shift.rating,
+            'worked': shift.worked,
+            'comment': shift.comment,
+            'worked_label': shift.worked_label,
+            'attendee_id': shift.attendee.id,
+            'attendee_name': shift.attendee.full_name
+        } for shift in job.shifts]
     }
 
 
+def update_counts(job, counts):
+    counts['all_total'] += job.total_hours
+    counts['all_signups'] += job.weighted_hours * len(job.shifts)
+    if job.restricted:
+        counts['restricted_total'] += job.total_hours
+        counts['restricted_signups'] += job.weighted_hours * len(job.shifts)
+    else:
+        counts['regular_total'] += job.total_hours
+        counts['regular_signups'] += job.weighted_hours * len(job.shifts)
```

Comment:
This file had a few places where we had to do some actual refactoring to use a more efficient strategy rather than just changing the query load strategy.  This is one example.